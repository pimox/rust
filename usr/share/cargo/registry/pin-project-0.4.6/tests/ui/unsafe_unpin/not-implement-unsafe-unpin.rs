use pin_project::pin_project;

#[pin_project(UnsafeUnpin)]
struct Foo<T, U> {
    #[pin]
    inner: T,
    other: U,
}

fn is_unpin<T: Unpin>() {}

fn main() {
    is_unpin::<Foo<(), ()>>(); //~ ERROR E0277
}
