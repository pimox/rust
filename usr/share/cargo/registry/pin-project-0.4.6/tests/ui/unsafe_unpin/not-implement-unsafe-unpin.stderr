error[E0277]: the trait bound `Foo<(), ()>: pin_project::UnsafeUnpin` is not satisfied
  --> $DIR/not-implement-unsafe-unpin.rs:13:16
   |
10 | fn is_unpin<T: Unpin>() {}
   |    --------    ----- required by this bound in `is_unpin`
...
13 |     is_unpin::<Foo<(), ()>>(); //~ ERROR E0277
   |                ^^^^^^^^^^^ the trait `pin_project::UnsafeUnpin` is not implemented for `Foo<(), ()>`
   |
   = note: required because of the requirements on the impl of `pin_project::UnsafeUnpin` for `pin_project::__private::Wrapper<'_, Foo<(), ()>>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Foo<(), ()>`
